include/solvers/mpm_base.tcc

```diff
@@ -92,8 +92,39 @@ mpm::MPMBase<Tdim>::MPMBase(const std::shared_ptr<IO>& io) : mpm::MPM(io) {
     abort();
   }
 
-  bool vtk_statevar = false;
+  // VTK particle variables
+  // Initialise container with empty vector
+  vtk_vars_.insert(
```

Comment:
These 3 lines can be in the constructor?

Reply:
Aren't these lines located in the constructor of MPMBase now?

Reply:
If they are in the constructor, why do we need it again here?

Reply:
What I mean is that these lines are written in the MPMBase constructor, see: https://github.com/cb-geo/mpm/blob/da3ba554b18d4d1a7e3757a67be28cab31e24884/include/solvers/mpm_base.tcc#L1-L103

Reply:
My question is why are reinitializing?

Reply:
`vtk_vars_` was initially empty, it is a map of `VariableType` and `std::vector<std::string>`. We are not reinitializing. We just insert an empty `std::vector<std::string>()` associated to the `VariableType` key. By doing so, the output can loop over those keys as:
https://github.com/cb-geo/mpm/blob/da3ba554b18d4d1a7e3757a67be28cab31e24884/include/solvers/mpm_base.tcc#L553
or:
https://github.com/cb-geo/mpm/blob/da3ba554b18d4d1a7e3757a67be28cab31e24884/include/solvers/mpm_base.tcc#L574
Without those three lines, `vtl_vars_.at(VariableType)` cant be called as it has not been initialized.

Reply:
My question is why do we have to do it again here if it's already in the constructor?
```
 // VTK particle variables 
   // Initialise container with empty vector 
   vtk_vars_.insert( 
       std::make_pair(mpm::VariableType::Scalar, std::vector<std::string>())); 
   vtk_vars_.insert( 
       std::make_pair(mpm::VariableType::Vector, std::vector<std::string>())); 
   vtk_vars_.insert( 
       std::make_pair(mpm::VariableType::Tensor, std::vector<std::string>())); 
```

Reply:
Sorry @kks32, I don't quite get what you are intending to say. These lines are written in the constructor. Line 97-102 are within the constructor of `MPMBase`, we only do those lines once:
```
//! Constructor 
 template <unsigned Tdim> 
 mpm::MPMBase<Tdim>::MPMBase(const std::shared_ptr<IO>& io) : mpm::MPM(io) { ... }
```